soh/soh/resource/importer/AudioSampleFactory.cpp

```diff
@@ -2,28 +2,95 @@
 #include "soh/resource/type/AudioSample.h"
 #include "spdlog/spdlog.h"
 
+#define DRMP3_IMPLEMENTATION
+#include <dr_libs/mp3.h>
+#define DRWAV_IMPLEMENTATION
+#include <dr_libs/wav.h>
+
+#define CODEC_S16 5
+
 namespace LUS {
 std::shared_ptr<IResource>
 AudioSampleFactory::ReadResource(std::shared_ptr<ResourceInitData> initData, std::shared_ptr<BinaryReader> reader) {
     auto resource = std::make_shared<AudioSample>(initData);
     std::shared_ptr<ResourceVersionFactory> factory = nullptr;
 
     switch (resource->GetInitData()->ResourceVersion) {
-    case 2:
-	factory = std::make_shared<AudioSampleFactoryV0>();
-	break;
+        case 0:
+            factory = std::make_shared<AudioSampleFactoryV0>();
+            break;
+        case 2:
+            factory = std::make_shared<AudioSampleFactoryV0>();
+            break;
+        case 0xC0:
+            factory = std::make_shared<AudioCustomSampleFactoryV0>();
+            break;
     }
 
     if (factory == nullptr) {
         SPDLOG_ERROR("Failed to load AudioSample with version {}", resource->GetInitData()->ResourceVersion);
-	return nullptr;
+	    return nullptr;
     }
 
     factory->ParseFileBinary(reader, resource);
 
     return resource;
 }
 
+void LUS::AudioCustomSampleFactoryV0::ParseFileBinary(std::shared_ptr<BinaryReader> reader,
```

Comment:
Maybe I misunderstand how these are supposed to be set up, but this doesn't seem consistent with the current patterns around supporting both binary and XML reading of resources. I would think this would be a ParseFileWav and ParseFileMP3 on the AudioSampleFactoryV0

Reply:
This is not the case because the header differs, putting everything on that factory is going to make things harder